### PR TITLE
Update of three badly terminated macro definitions

### PR DESCRIPTION
**Pull request by newbrain**
Description
-----------
The following three macros in `task.h` were written in a way that forced to to omit the terminating semicolon when used in a non-block, single statement, `then` part of a if statement:
```
if(condition)
    macro(args, more, args);
else
   /* Any block or statement */
```
Compilers would complain of a syntax error, as the terminating semicolon or the block ending was already part of the macro.
- vTaskDelayUntil() was updated to conform to usual pattern do { ... } while(0), making it into an unterminated statement.
- vTaskNotifyGiveFromISR() and
- vTaskGenericNotifyGiveFromISR() were updated to remove the unneeded terminating semicolons.

I scanned the other files in the include directory for similar patterns and found none.

Test Steps
-----------

Attempts to compile any code similar to what described above will no longer fail.

Related Issue
-----------

- This PR addresses issues #553 and #554
- The same issue is present in the `smp`branch and in the latest LTS release.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by paulbartell — DISMISSED** [review]

**Review by paulbartell — APPROVED** [review]

**Comment by paulbartell** [comment]
@newbrain Thanks very much for the PR. I will get a second review and merge this in the morning.

**Comment by newbrain** [comment]
Thanks! Sorry for missing the formatting rule in `while( 0 )`!

**Review by AniruddhaKanhere — APPROVED** [review]

**Comment by AniruddhaKanhere** [comment]
Thanks @newbrain for this PR.
I have approved this and will be merging this in :)

**Pull request merged**
